### PR TITLE
Expose parameters for H.264 encoder

### PR DESCRIPTION
**Pull request by fiban-havok**

### CODE REVIEW

File: third_party/winuwp_h264/H264Encoder/H264Encoder.cc [hunk 1]
```diff
@@ -204,12 +244,47 @@ int WinUWPH264EncoderImpl::InitWriter() {
...
+      default:
+        RTC_NOTREACHED();
+    }
+
+    ON_SUCCEEDED(encodingAttributes->SetUINT32(
+        CODECAPI_AVEncCommonRateControlMode, mode));
+  }
+
+  // Enable CABAC. This will only have effect in profiles that support it.
+  ON_SUCCEEDED(encodingAttributes->SetUINT32(CODECAPI_AVEncH264CABACEnable,
```
Comment (jehumb-havok) [inline third_party/winuwp_h264/H264Encoder/H264Encoder.cc:275]: Can you log a message for that maybe? Also no idea what CABAC is, so maybe add 1-line explanation why we hard-code enable it?

File: third_party/winuwp_h264/H264Encoder/H264Encoder.cc [hunk 2]
```diff
@@ -473,13 +548,6 @@ int WinUWPH264EncoderImpl::Encode(
 
   ON_SUCCEEDED(sinkWriter_->WriteSample(streamIndex_, sample.Get()));
 
-  rtc::CritScope lock(&crit_);
-  // Some threads online mention this is useful to do regularly.
-  ++frameCount_;
-  if (frameCount_ % 30 == 0) {
-    ON_SUCCEEDED(sinkWriter_->NotifyEndOfSegment(streamIndex_));
```
Comment (jehumb-havok) [inline third_party/winuwp_h264/H264Encoder/H264Encoder.cc:480]: Why?

**Review by jehumb-havok — APPROVED** [review]

File: third_party/winuwp_h264/H264Encoder/H264Encoder.cc (hunk 2 above)
Comment (fiban-havok) [reply third_party/winuwp_h264/H264Encoder/H264Encoder.cc:480]: See commit message
> Unjustified and might even make things worse.

File: third_party/winuwp_h264/H264Encoder/H264Encoder.cc (hunk 1 above)
Comment (fiban-havok) [reply third_party/winuwp_h264/H264Encoder/H264Encoder.cc:275]: Done

**Pull request merged**